CMakeLists.txt

```diff
@@ -1,26 +1,82 @@
-cmake_minimum_required (VERSION 3.14)
-project (ifc-sdk)
+# Copyright Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-# Fetch Microsoft GSL
-include(FetchContent)
+cmake_minimum_required(VERSION 3.14)
 
-FetchContent_Declare(GSL
-    GIT_REPOSITORY "https://github.com/microsoft/GSL"
-    GIT_TAG "v4.0.0"
+project(
+    ifc-sdk
+    VERSION 0.0.1
+    LANGUAGES CXX
 )
-FetchContent_MakeAvailable(GSL)
 
-if (MSVC)
-add_compile_options(
-    /EHsc          # Turn on exception handling semantics
+option(DEVELOPER_MODE "Enable project options and targets for developers of ifc-sdk" OFF)
+mark_as_advanced(DEVELOPER_MODE)
+include(CMakeDependentOption)
+
+find_package(Microsoft.GSL REQUIRED)
+
+add_library(
+    ifc-reader STATIC
+    src/file.cxx
+    src/sgraph.cxx
+    src/ifc-reader/operators.cxx
+    src/ifc-reader/reader.cxx
+    src/ifc-reader/util.cxx
 )
+add_library(Microsoft.IFC::ifc-reader ALIAS ifc-reader)
+if(WIN32)
+  target_compile_definitions(ifc-reader PRIVATE NOMINMAX)
+  target_link_libraries(ifc-reader PRIVATE bcrypt)
+  target_sources(ifc-reader PRIVATE src/hash_win.cxx)
+else()
+  target_sources(ifc-reader PRIVATE src/sha256.cxx)
 endif()
+target_link_libraries(ifc-reader PUBLIC Microsoft.GSL::GSL)
+target_compile_features(ifc-reader PUBLIC cxx_std_20)
+target_include_directories(ifc-reader PUBLIC "\$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>")
```

Comment:
Not really. Why is `/std:c++latest` necessary?

Reply:
We want to get into the habit of the project building at the tip of the language (C++23 for now; C++26 when the committee votes out the C++26 standard, etc.)

Reply:
There is no way to cleanly do that and without knowing more that doesn't look like a sound plan. You can always just make a new release when it feels appropriate. [CMAKE_CXX_KNOWN_FEATURES](https://cmake.org/cmake/help/latest/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html) tracks what version of CMake started supporting each standard.

Reply:
So if the minimum required version was bumped to 3.27 (at least 3.20 according to the doc), it would be possible to request C++23?  We can make that change along with the CMake version bump in a separate PR if those changes don't fit here.